spec.bs

```diff
@@ -686,11 +704,40 @@ to include them in subsequent requests to the origins that can verify its
 validity. An issuer might choose to include an optional <a http-header>Sec-Private-State-Token-Lifetime</a>
 header in the redemption response. The value of this header indicates the
 expiration time for the [=redemption record=] provided. This expiration is
-specified as number of seconds in the <a http-header>Sec-Private-State-Token-Lifetime</a> HTTP response
+specified as number of seconds in the <a http-header>`Sec-Private-State-Token-Lifetime`</a> HTTP response
 header value.
 
 A <dfn>Redemption Record</dfn> is a [=byte sequence=].
 
+Private State Tokens API provides `'send-redemption-record'` operation to [=append private state token redemption record headers=]. This operation attaches a previously [=record a redemption record=] from [=handle a redeem response=].
+
+To <dfn>append private state token redemption record headers</dfn> given a [=/request=] |request|, run the following steps:
+
+  1. If |request|'s [=request/client=] is not a [=secure context=], then abort these steps.
+  1. Let |topLevel| be |request|'s [=request/client=]'s [=environment/top-level origin=].
+  1. [=list/For each=] |issuer| of [=request/issuers=]:
+    1. Let |issuerURL| be the the result of running the [=URL parser=] on |issuer|.
+    1. If |issuerURL| is failure, then abort these steps.
+    1. If |issuerURL|'s [=url/scheme=] is not an [=HTTP(S) scheme=], then abort these steps.
+    1. Let |issuerOrigin| be |issuerURL|'s [=/origin=].
+    1. If |issuerOrigin| is not a [=potentially trustworthy origin=], then abort these steps.
+  1. Let |records_per_issuer| be a [=map=] where keys are USVString and values are [=redemption records=].
+  1. [=list/For each=] |issuer| of [=request/issuers=]:
+    1. Let |record| be the result of performing [=retrieve a redemption record=] with |issuer| and |topLevel|.
+    1. If |record| is null, then [=iteration/continue=].
+    1. [=map/Set=] |records_per_issuer|[|issuer|] to |record|.
+  1. If |records_per_issuer| is empty, then abort these steps.
+  1. Let |headerItems| be a structured headers list [[RFC8941]].
+  1. [=map/For each=] |issuer| -> |record| of |records_per_issuer|:
+    1. Let |serializedIssuer| be result of serializing |issuer|.
+    1. Let |serializedRecord| be result of serializing |record|.
+    1. [=list/Append=] |serializedIssuer| and |serializedRecord| pairs to |headerItems|.
```

Comment:
I'm not an expert on structured headers but instead of "pairs" I think you need to construct an [inner list](https://www.rfc-editor.org/rfc/rfc8941.html#name-inner-lists) here?